zerver/lib/bugdown/__init__.py

```diff
@@ -522,22 +522,17 @@ def run(self, root: Element) -> None:
                 continue
             img.set("src", get_camo_url(url))
 
-class BacktickPattern(markdown.inlinepatterns.Pattern):
+class BacktickInlineProcessor(markdown.inlinepatterns.BacktickInlineProcessor):
     """ Return a `<code>` element containing the matching text. """
-
-    def __init__(self, pattern: str) -> None:
-        markdown.inlinepatterns.Pattern.__init__(self, pattern)
-        self.ESCAPED_BSLASH = '{}{}{}'.format(markdown.util.STX, ord('\\'), markdown.util.ETX)
-        self.tag = 'code'
-
-    def handleMatch(self, m: Match[str]) -> Union[str, Element]:
-        if m.group(4):
-            el = Element(self.tag)
-            # Modified to not strip whitespace
-            el.text = markdown.util.AtomicString(m.group(4))
-            return el
-        else:
-            return m.group(2).replace('\\\\', self.ESCAPED_BSLASH)
+    def handleMatch(self, m: Match[str], data: str) -> Tuple[Union[None, Element], int, int]:
+        # Let upstream's implementation do its job as it is, we'll just replace the
+        # text to not strip the group
+        el, start, end = super().handleMatch(m, data)
+        if m.group(3):
```

Comment:
Is there a more robust way than `m.group(3)` to get at this?

Reply:
No, here's the upstream:

```py
def handleMatch(self, m, data):
        if m.group(3):
            el = etree.Element(self.tag)
            el.text = util.AtomicString(util.code_escape(m.group(3).strip()))
            return el, m.start(0), m.end(0)
        else:
            return m.group(1).replace('\\\\', self.ESCAPED_BSLASH), m.start(0), m.end(0)
```

I'm also reusing that `if` logic here, which I wish we didn't have to, but its still better than having all of the code here.